blackjax/adaptation/pathfinder_adaptation.py

```diff
@@ -18,109 +17,110 @@
-        pathfinder_state = pathfinder_init_fn(
-            pathfinder_rng_key, logprob_fn, initial_position
-        )
-        new_initial_position, _ = sample_from_state(sample_rng_key, pathfinder_state)
+        """
+        pathfinder_key, sample_key = jax.random.split(rng_key, 2)
+        pathfinder_state = pathfinder_init_fn(pathfinder_key, logprob_fn, position)
+        new_position, _ = sample_from_state(sample_key, pathfinder_state)
         inverse_mass_matrix = lbfgs_inverse_hessian_formula_1(
             pathfinder_state.alpha, pathfinder_state.beta, pathfinder_state.gamma
         )
 
-        warmup_state = PathfinderAdaptationState(da_state, inverse_mass_matrix)
+        da_state = da_init(initial_step_size)
+
+        warmup_state = PathfinderAdaptationState(
+            da_state, initial_step_size, inverse_mass_matrix
+        )
 
-        return warmup_state, new_initial_position
+        return warmup_state, new_position
 
     def update(
-        rng_key: PRNGKey,
-        chain_state: HMCState,
         adaptation_state: PathfinderAdaptationState,
-    ) -> Tuple[HMCState, PathfinderAdaptationState, NamedTuple]:
-        """Move the warmup by one step.
+        position: PyTree,
+        acceptance_rate: float,
+    ) -> PathfinderAdaptationState:
+        """Update the adaptation state and parameter values.
 
-        We first create a new kernel with the current values of the step size
-        and mass matrix and move the chain one step. Then, we update the dual
-        averaging adaptation algorithm.
+        Since the value of the inverse mass matrix is already known we only
+        update the step size state.
 
         Parameters
         ----------
-        rng_key
-            The key used in JAX's random number generator.
-        chain_state
-            Current state of the chain.
-        adaprtation_state
-            Current warmup state.
+        adaptation_state
+            Current adptation state.
+        adaptation_stage
+            The current stage of the warmup: whether this is a slow window,
+            a fast window and if we are at the last step of a slow window.
+        position
+            Current value of the model parameters.
+        acceptance_rate
+            Value of the acceptance rate for the last mcmc step.
 
         Returns
         -------
         The updated states of the chain and the warmup.
 
         """
-        step_size = jnp.exp(adaptation_state.da_state.log_step_size)
-        inverse_mass_matrix = adaptation_state.inverse_mass_matrix
-        kernel = kernel_factory(step_size, inverse_mass_matrix)
-
-        chain_state, chain_info = kernel(rng_key, chain_state)
-        new_da_state = da_update(adaptation_state.da_state, chain_info)
-        new_warmup_state = PathfinderAdaptationState(
-            new_da_state, adaptation_state.inverse_mass_matrix
-        )
+        new_ss_state = da_update(adaptation_state.ss_state, acceptance_rate)
```

Comment:
Amazing - this is much better than initializing a kernel and calling it inside adaptation.

Reply:
Glad you like it :)